arca/backend/docker.py

```diff
@@ -0,0 +1,271 @@
+import itertools
+import json
+import sys
+import tarfile
+import time
+
+from io import BytesIO
+from pathlib import Path
+from typing import Optional
+
+import docker
+from docker.errors import BuildError, APIError
+from docker.models.containers import Container
+from docker.models.images import Image
+from requests.exceptions import ConnectionError
+
+import arca
+from arca.result import Result
+from arca.task import Task
+from arca.utils import logger, LazySettingProperty
+from .base import BaseBackend
+
+
+class DockerBackend(BaseBackend):
+
+    python_version = LazySettingProperty(key="python_version", default=None)
+    keep_container_running = LazySettingProperty(key="keep_container_running", default=False)
+    apk_dependencies = LazySettingProperty(key="apk_dependencies", default=None)
+    disable_pull = LazySettingProperty(key="disable_pull", default=False)  # so the build can be tested
+
+    def __init__(self, **kwargs):
+        super(DockerBackend, self).__init__(**kwargs)
```

Comment:
Are you aiming for Python 2 compatibility? If not, maybe just use `super()`.

Reply:
Oh, nope, not at all, replaced, PyCharm autocompletion did that...